docs/actions-scripts.md

```diff
@@ -1,23 +1,103 @@
 The most common kind of action is a scripted action.
 
 Generally-speaking, you can write whatever code you like as long as it will run successfully on server, and it is possible to [test this locally](actions-pipelines.md#running-your-code-locally).
+
 However, note the following restrictions and guidance:
 
 * **Write analyses in Python, R, or Stata.**
 You can can use more than one language in a single project if necessary.  You can find more information about the available libraries below.
+
 * **Do not write code that requires an internet connection to run.**
 Any research objects (datasets, libraries, etc.) that are retrieved via the internet should be imported to the repo locally first.
 If this is not possible (for instance if the object size is too large to be transferred via GitHub) then get in touch.
+
 * **Avoid code that consumes a lot of time or memory.** The server is not an infinite resource. We can advise on code optimisation if run-times become problematic.  A good strategy is to split your processing into separate project pipeline actions; the job runner can then choose to run them in parallel if sufficient resources are available.
-* **Write code that runs across different platforms.**
-Since code will be run both locally and within a Linux-based Docker environment. In practice, this just means ensuring you use forward-slashes `/` for directories.
+
+* **Write code that runs in the OpenSAFELY platform.**
+Code will be run within a Linux-based Docker environment. In practice, this just means ensuring you use forward-slashes `/` for directories.
+
 * **Structure your code into discrete chunks, both within scripts, and by splitting into different pipeline actions.**
 This helps with:
 	* readability
 	* bug-finding
 	* parallelisation via the project pipeline
 
 
+## Reading and Writing Outputs
+
+Scripted actions can read and write output files that are saved in the workspace. These generally fall into two categories: large files of `highly_sensitive` data to feed into other actions, or smaller `moderately_sensitive` outputs intended for review and release.
+
+
+### Large `highly_sensitive` Output Files
+
+It is important that the right files formats are used for large data files. The wrong formats can waste disk space, execution time, and server memory. The specific formats used vary with language ecosystem, but they should always be compressed.
+
+Note: the `cohortextractor` command produces `csv.gz` outputs, which is the current recommended output to be consumed by all supported languages.
```

Comment:
suggestion:

Move this to an admonition:

```suggestion
!!! note
    The `cohortextractor` command produces `csv.gz` outputs. `csv.gz` is the recommended output format.
```

I'd also be inclined to link to an explanation of:

> `csv.gz` is the recommended output format.

but we don't have a definitive section. It's part of the `docs/actions-cohortextractor.md` content, but not in its own separate section. This isn't too important though.